src/features/unassigned-list/unassigned-list.js

```diff
@@ -0,0 +1,133 @@
+import { newUserSelector } from '@/models/index.js';
+import className from 'classnames';
+import { Cookie } from '../../utils/index.js';
+import {addMentor, createSecretary, addStudent } from '@/models/index.js';
+
+import Icon from '../../icon.js';
+import { Search, Button, WithLoading } from '../../components/index.js';
+
+import { fetchUnAssignedUserList } from '../../models/index.js';
+
+import styles from './unassigned-list.scss';
+
+export const UnAssignedList = () => {
+  const roles = ['student', 'mentor', 'secretery'];
+  const currentUserRole = JSON.parse(Cookie.get('user') ?? null).role;
+  const { loaded, notAssigned } = useSelector(newUserSelector);
+
+  const [getUnAssignedUserList] = useActions([fetchUnAssignedUserList]);
+
+  const [addStudentRole,
+    addSecreteryRole,
+    addMentorRole] = useActions([addStudent, createSecretary, addMentor]);
+
+  const [usersRoles, setUsersRole] = useState(notAssigned[0]?.map((user) => ({ id: user.id, role: 1 })));
+
+  const [search, setSearch] = useState('');
+  const [searchPersonValue, setSearchPersonValue] = useState([]);
+
+  useEffect(() => {
+    getUnAssignedUserList();
+  }, [getUnAssignedUserList]);
+
+  useEffect(() => {
+    if (loaded) {
+      setSearchPersonValue(notAssigned[0]);
+    }
+  }, [notAssigned[0]]);
+
+  useEffect(() => {
+    if (loaded) {
+      const results = notAssigned[0]?.filter((user) => (
+        (user.firstName.concat(user.lastName)).toUpperCase())
+        .includes(search.toUpperCase()));
+      setSearchPersonValue(results);
+    }
+  }, [search]);
+
+  const changeRole = (id, value) => {
+    const newState = searchPersonValue.map((user) => (user.id === id ? ({ ...user, role: Number(value) }) : user));
+    setUsersRole(newState);
+  };
+
+  const handleButtonClick = (id) => {
+    const { role } = usersRoles.find((user) => user.id === id);
+    if (role !== 0) {
+      switch (role) {
+        case 1:
+          return addStudentRole(id);
+        case 2:
+          return addMentorRole(id);
+        case 3:
+          return addSecreteryRole(id);
+        default: return {};
+      }
+    }
+  };
+
+  const handleSearch = (inputValue) => {
+    setSearch(inputValue);
+  };
+
+  const options = () => {
+    switch (currentUserRole) {
+      case 2:
+        return <option value={1}>{roles[0]}</option>;
+      case 3:
+        return roles.slice(0, 2).map((role) => (
+          <option value={roles.indexOf(role) + 1}>
+            {role}
+          </option>
+        ));
+      case 4:
+        return roles.map((role) => (
+          <option value={roles.indexOf(role) + 1}>
+            {role}
+          </option>
+        ));
+      default: return {};
+    }
+  };
+
+  const list = () => {
+    if (loaded) {
+      if (searchPersonValue.length !== 0) {
+        return (searchPersonValue.map((user) => (
+          <div className={styles.card}>
+            <p>{`${user.firstName} ${user.lastName}`}<br />{user.email}</p>
```

Comment:
No need to use template strings, you can simplify: 
`<p>{user.firstName} {user.lastName}<br />{user.email}</p>`